LayoutTests/http/tests/notifications/notification-permission-query.html

```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/js-test-pre.js"></script>
+<script src="/resources/notifications-test-pre.js"></script>
+</head>
+<body>
+<script>
+
+/*
+ * This test tests that Permissions::query() retrieves the true permission state of the Notifications API
+ * in all 3 cases: no permission state has been set (default), the permission state is granted, and the
+ * permission state is denied. 
+ */
+
+description("This test checks that Permissions::query() returns the true permission state of the Notifications API.")
+
+jsTestIsAsync = true;
+
+function defaultTest()
+{
+    testRunner.removeAllWebNotificationPermissions();
```

Comment:
What's this? This shouldn't be needed. We don't rely on tests to clear state at the beginning or at the end of the test. It is the WebKitTestRunner's job.

Reply:
Fixed!